services/frontend/adapters/inbound/k8s_sidecar_notifications_receiver.adapter.go

```diff
@@ -0,0 +1,131 @@
+package inboundAdapters
+
+import (
+	"context"
+	"fmt"
+	"net"
+	"net/http"
+	"os"
+	"sync"
+	"time"
+
+	"github.com/rs/zerolog/log"
+
+	"github.com/berops/claudie/services/frontend/domain/usecases"
+)
+
+// Receives notifications from the k8s-sidecar container running in the same pod
+// The notifications are regarding new/updated claudie manifests
+type K8sSidecarNotificationsReceiver struct {
+	usecases *usecases.Usecases
+
+	// manifestDir represents the path to the directory containing claudie manifest files which will be
+	// watched by the k8s-sidecar container.
+	manifestDir string
+
+	// server is the underlying HTTP server which receives notifications
+	// from the k8s sidecar container when the manifest files are created/updated/deleted.
+	server *http.Server
+
+	// waitGroup is used to handle a graceful shutdown of the HTTP server.
+	// It will wait for all spawned go-routines to finish their work.
+	waitGroup sync.WaitGroup
+}
+
+// NewK8sSidecarNotificationsReceiver creates an instance of the K8sSidecarNotificationsReceiver struct.
+// It registers the notification handling route to the underlying HTTP server.
+// Then performs an initial healthcheck of K8sSidecarNotificationsReceiver.
+func NewK8sSidecarNotificationsReceiver(usecases *usecases.Usecases) (*K8sSidecarNotificationsReceiver, error) {
+
+	manifestDir, isEnvFound := os.LookupEnv("MANIFEST_DIR")
+	if !isEnvFound {
+		return nil, fmt.Errorf("env MANIFES_DIR not found")
+	}
+
+	k8sSidecarNotificationsReceiver := &K8sSidecarNotificationsReceiver{
+
+		manifestDir: manifestDir,
+		usecases:    usecases,
+
+		server: &http.Server{ReadHeaderTimeout: 2 * time.Second},
+	}
+
+	k8sSidecarNotificationsReceiver.registerNotificationHandlers()
+
+	go k8sSidecarNotificationsReceiver.watchConfigs()
+
+	return k8sSidecarNotificationsReceiver, k8sSidecarNotificationsReceiver.PerformHealthCheck()
+}
+
+// registerNotificationHandlers registers a router to the underlying HTTP server.
+// The router contains a route ("/reload") that handles incoming notifications from the K8s-sidecar.
+func (k *K8sSidecarNotificationsReceiver) registerNotificationHandlers() {
+
+	var router *http.ServeMux = http.NewServeMux()
+
+	// TODO: rename the route "/reload" to "/process-manifest-files"
```

Comment:
Is this planned ? if not remove the `TODO`

If yes, rename it in the manifest as well https://github.com/berops/claudie/blob/065c07e714e392be35a7afe9b8887fdc3566b508/manifests/claudie/frontend.yaml#L82